app/src/main/java/com/example/cartoon/Sp_screen.kt

```diff
@@ -20,42 +18,40 @@ class Sp_Screen : AppCompatActivity() {
         val animatedText = findViewById<TextView>(R.id.animatedText)
         val animatedImage = findViewById<ImageView>(R.id.img)
 
-        val imageAnimation = ObjectAnimator.ofFloat(
-            animatedImage,
-            "translationY",
-            0f,
-            -200f // Adjust the value to determine how much the image moves up
-        )
-        imageAnimation.duration = 1000 // 1 second
-        imageAnimation.interpolator = DecelerateInterpolator()
-
-        // Start the image animation with a slight delay
-        Handler().postDelayed({
-            imageAnimation.start()
-        }, IMAGE_ANIMATION_DELAY)
-
+        // Image animation
+        startAnimation(animatedImage, "translationY", 0f, -200f, 1000, DecelerateInterpolator())
 
         // Text animation
-        val fadeInAnimation = ObjectAnimator.ofFloat(animatedText, "alpha", 0f, 1f)
-        fadeInAnimation.duration = 1000 // 1 second
-        fadeInAnimation.interpolator = AccelerateDecelerateInterpolator()
+        startAnimation(animatedText, "alpha", 0f, 1f, 1000, AccelerateDecelerateInterpolator())
 
-        // Start the text animation with a slight delay
+        // Transition to the main activity after a delay
         Handler().postDelayed({
-            fadeInAnimation.start()
-        }, TEXT_ANIMATION_DELAY)
-
-        // Delay for a few seconds and then transition to the main activity
-        Handler().postDelayed({
-            val intent = Intent(this, MainActivity::class.java)
-            startActivity(intent)
+            startActivity(Intent(this, MainActivity::class.java))
             finish()
         }, SPLASH_DELAY)
     }
 
     companion object {
-        private const val IMAGE_ANIMATION_DELAY: Long = 1000 // 1 second (adjust as needed)
         private const val SPLASH_DELAY: Long = 3000 // 3 seconds
-        private const val TEXT_ANIMATION_DELAY: Long = 500 // 0.5 seconds (adjust as needed)
+    }
+
+    private fun startAnimation(
+        view: ImageView, property: String, startValue: Float, endValue: Float,
+        duration: Long, interpolator: DecelerateInterpolator
+    ) {
+        val animation = ObjectAnimator.ofFloat(view, property, startValue, endValue)
+        animation.duration = duration
+        animation.interpolator = interpolator
+        Handler().postDelayed({ animation.start() }, IMAGE_ANIMATION_DELAY)
+    }
+
+    private fun startAnimation(
+        view: TextView, property: String, startValue: Float, endValue: Float,
+        duration: Long, interpolator: AccelerateDecelerateInterpolator
+    ) {
+        val animation = ObjectAnimator.ofFloat(view, property, startValue, endValue)
+        animation.duration = duration
+        animation.interpolator = interpolator
+        Handler().postDelayed({ animation.start() }, TEXT_ANIMATION_DELAY)
```

Comment:
Handler().postDelayed({ animation.start() }, TEXT_ANIMATION_DELAY)

Unresolved reference: TEXT_ANIMATION_DELAY

Reply:
Good to have a reply from you @ashut0sh75 I will resolve it and will give my contribution according to guidelines